group/assignment6/Challenge3.java

```diff
@@ -0,0 +1,63 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+public class Challenge3 {
+    /**
+     * We use arrays to represent parking slots as this is the simplest DS and most convieniet to use and numbers denote
+     * car IDs and we assume that there is no duplicate car ID e.g elements are always unique and we denote empty car space
+     * slot a special number '0'.
+     * The basic idea is to loop over the cars one by one and to see if it's not in the desired final place then we try to
+     * look for the empty slot '0' and swap it with the car and put the desired car in the current slot. The algorithm takes
+     * O(n*n) one pass for going over the cars and another pass for trying to find index of 0 but I believe can be enhanced using
+     * hashmaps.
+     * Reserved uptade: When the current car cannot be placed to empty slot, it iterates over other cars until it find a
+     * placeable car, although it is in the right place.
+     * @param source initial arrangement of cars in the parking
+     * @param destination desired arrangement of cars in the parking.
+     */
+    public static void generateMoves(int source[], int destination[],
```

Comment:
Very cool!

Do you think you could try to integrate this to RearrangeCar class instead of having a new class? Because there is a lot of code that could be nicely reused.. this method could be simply imported by removing the arguments `source` and `destination` and using `startConfiguration` and `endConfiguration` instead 😁 )